contracts/cw20-streams/src/contract.rs

```diff
@@ -0,0 +1,575 @@
+            Config {
+                owner: Addr::unchecked("creator"),
+                cw20_addr: Addr::unchecked(MOCK_CONTRACT_ADDR)
+            }
+        );
+    }
+
+    #[test]
+    fn execute_withdraw() {
+        let mut deps = mock_dependencies();
+        let msg = InstantiateMsg {
+            owner: None,
+            cw20_addr: String::from(MOCK_CONTRACT_ADDR),
+        };
+        let info = mock_info(MOCK_CONTRACT_ADDR, &[]);
+        instantiate(deps.as_mut(), mock_env(), info.clone(), msg).unwrap();
+
+        let sender = Addr::unchecked("Alice").to_string();
+        let recipient = Addr::unchecked("Bob").to_string();
+        let amount = Uint128::new(200);
+        let env = mock_env();
+        let start_time = env.block.time.plus_seconds(100).seconds();
+        let end_time = env.block.time.plus_seconds(300).seconds();
+
+        let msg = ExecuteMsg::Receive(Cw20ReceiveMsg {
+            sender,
+            amount,
+            msg: to_binary(&ReceiveMsg::CreateStream {
+                recipient,
+                start_time,
+                end_time,
+            })
+            .unwrap(),
+        });
+        execute(deps.as_mut(), env.clone(), info, msg).unwrap();
+
+        let msg = QueryMsg::GetStream { id: 1 };
+        let res = query(deps.as_ref(), mock_env(), msg).unwrap();
+        let stream: Stream = from_binary(&res).unwrap();
+
+        assert_eq!(
+            stream,
+            Stream {
+                owner: Addr::unchecked("Alice"),
+                recipient: Addr::unchecked("Bob"),
+                amount,
+                claimed_amount: Uint128::new(0),
+                start_time,
+                rate_per_second: Uint128::new(1),
+                end_time
+            }
+        );
+
+        // Stream has not started
+        let mut info = mock_info(MOCK_CONTRACT_ADDR, &[]);
+        info.sender = Addr::unchecked("Bob");
+        let msg = ExecuteMsg::Withdraw { id: 1 };
+        let err = execute(deps.as_mut(), env, info, msg).unwrap_err();
+        assert_eq!(err, ContractError::NoFundsToClaim {});
+
+        // Stream has started so tokens have vested
+        let msg = ExecuteMsg::Withdraw { id: 1 };
+        let mut info = mock_info(MOCK_CONTRACT_ADDR, &[]);
+        let mut env = mock_env();
+        info.sender = Addr::unchecked("Bob");
+        env.block.time = env.block.time.plus_seconds(150);
+        let res = execute(deps.as_mut(), env, info, msg).unwrap();
+        let msg = res.messages[0].clone().msg;
+
+        assert_eq!(
+            msg,
+            CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: String::from(MOCK_CONTRACT_ADDR),
+                msg: to_binary(&Cw20ExecuteMsg::Transfer {
+                    recipient: String::from("Bob"),
+                    amount: Uint128::new(50)
+                })
+                .unwrap(),
+                funds: vec![]
+            })
+        );
+
+        let msg = QueryMsg::GetStream { id: 1 };
```

Comment:
Style note: I would make a helper for this pattern.
At least these three lines with a "get_stream()" test helper.

Then:

```rust
let expected = Stream {
                 owner: Addr::unchecked("Alice"),
                 recipient: Addr::unchecked("Bob"),
                 amount,
                 claimed_amount: Uint128::new(50),
                 start_time,
                 rate_per_second: Uint128::new(1),
                 end_time
             };
assert_eq!(expected, get_stream(&deps, 1));
```

We see it a lot of places and removing boilerplate in tests makes it easier to see the actual flow being tested